src/features/content-answers/ContentAnswersModalContent.tsx

```diff
@@ -1,38 +1,77 @@
 import React from 'react';
 
-import Answer from './Answer';
-import Question from './Question';
-import WelcomeMessage from './WelcomeMessage';
-
-import { QuestionType } from './ContentAnswersModal';
+import ContentAnswersModalChat from './ContentAnswersModalChat';
 // @ts-ignore: no ts definition
 // eslint-disable-next-line import/named
 import { User } from '../../../common/types/core';
 
-import './ContentAnswersModalContent.scss';
+export type QuestionType = {
+    prompt: string;
+    answer?: string;
+    createdAt?: string;
+};
 
 type Props = {
-    currentUser: User;
+    currentUser?: User;
     fileName: string;
     isLoading: boolean;
     questions: QuestionType[];
 };
 
 const ContentAnswersModalContent = ({ currentUser, fileName, isLoading, questions }: Props) => {
+    const contentRef = React.useRef<HTMLDivElement | null>(null);
+    const messagesEndRef = React.useRef<HTMLDivElement | null>(null);
+    const userScrolled = React.useRef(false);
+
+    const handleScrollToBottom = React.useCallback(behavior => {
+        if (messagesEndRef.current && messagesEndRef.current.scrollIntoView && contentRef.current) {
+            // Only scroll if the content is at bottom
+            if (!userScrolled.current) {
+                messagesEndRef.current.scrollIntoView({ behavior });
+            }
+        }
+    }, []);
+
+    // Handle user scrolling
+    const handleScroll = () => {
+        if (contentRef.current) {
+            const { scrollTop, clientHeight, scrollHeight } = contentRef.current;
+            const threshold = scrollHeight - (scrollTop + clientHeight);
+            userScrolled.current = threshold > 5;
+        }
+    };
+
+    // Scroll to the bottom on chat load
+    React.useEffect(() => {
+        setTimeout(() => {
```

Comment:
Why setTimeout?

Reply:
There is [a known issue](https://stackoverflow.com/questions/71181018/scrollintoview-doesnt-scroll-anywhere) for scrollingIntoView function that will not scroll to anywhere when it is invoked before the changed state is rendered. And by using setTimeout here, it would help to put handleScrollToBottom function at the end of the event loop which prevents this above concern.